File: csharp/Example3.cs
using System;
using Rinearn.ExevalatorCS;

/// <summary>
/// An example to use a variable.
/// </summary>
static class Example3
{
    static void Main()
    {
        // Create an instance of Exevalator Engine
        Exevalator exevalator = new Exevalator();

        // Declare a variable and set the value
        exevalator.DeclareVariable("x");
        exevalator.WriteVariable("x", 1.25);

        // Evaluate the value of an expression
        double result = exevalator.Eval("x + 1");

        // Display the result
        Console.WriteLine("result: " + result);
    }
}

File: rust/example3.rs
mod exevalator;
use exevalator::Exevalator;

/// An example to use a variable.
fn main() {

    // Create an instance of Exevalator Engine
    let mut exevalator = Exevalator::new();

    // Declare a variable and set the value
    match exevalator.declare_variable("x") {
        Ok(address) => address,
        Err(declaration_error) => panic!("{}", declaration_error)
    };
    match exevalator.write_variable("x", 1.25) {
        Some(access_error) => panic!("{}", access_error),
        None => {},
    };

    // Evaluate the value of an expression
    let result: f64 = match exevalator.eval("x + 1") {
        Ok(eval_value) => eval_value,
        Err(eval_error) => panic!("{}", eval_error),
    };

    // Display the result
    println!("result: {}", result);    
}
